components/fastly_frontend/main.tf

```diff
@@ -18,11 +18,39 @@ variable "papertrail_destination" {
 }
 
 locals {
-  headers = {
+  request_headers = {
+    # Provide geolocation information to web servers:
     "X-Fastly-Country-Code" = "client.geo.country_code",
     "X-Fastly-Region-Code"  = "client.geo.region",
     "X-Fastly-Postal-Code"  = "client.geo.postal_code",
   }
+
+  response_headers = {
+    # Expose geolocation information in debug headers:
+    "X-Fastly-Country-Code" = "client.geo.country_code",
+    "X-Fastly-Region-Code"  = "client.geo.region",
+    "X-Fastly-Postal-Code"  = "client.geo.postal_code",
+
+    # Ensures that we don't leak sensitive items from URLs in Referrer headers: we will
+    # send full URL if same-origin and HTTPS, origin-only (e.g. 'www.dosomething.org')
+    # when crossing domains, and nothing at all when going to HTTP.
+    "Referrer-Policy" = "\"strict-origin-when-cross-origin\""
+
+    # Enforce HTTPS for all traffic (TODO: https://www.pivotaltracker.com/story/show/174911098)
+    "Strict-Transport-Security" = "\"max-age=300; includeSubDomains\"" # 5 minutes.
+
+    # Prevent sniffing content types:
+    "X-Content-Type-Options" = "\"nosniff\""
+
+    # Restrict this application from being embedded as an iframe on
+    # third-party domains, to prevent click-jacking attacks:
+    "X-Frame-Options" = "\"SAMEORIGIN\""
+
+    # Enable XSS filtering in some browsers. This should be
+    # replaced by a strong Content Security Policy in the future.
+    # TODO: https://www.pivotaltracker.com/story/show/174911196
+    "X-XSS-Protection" = "\"1\""
```

Comment:
These are all identical to what's set in the `fastly_backend` module above! I think it's reasonable to keep these not-so-DRY, since we may eventually have differing security needs for backends and frontends.